gammapy/modeling/models/spectral.py

```diff
@@ -213,6 +213,31 @@ def evaluate_error(self, energy, epsilon=1e-4):
         """
         return self._propagate_error(epsilon=epsilon, fct=self, energy=energy)
 
+    def pivot_energy(self, energy):
+        """The decorrelation energy for a given spectral model calculated numerically.
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+            Array of energies to evaluate the spectral model and error.
+
+        Returns
+        -------
+        pivot energy : `~astropy.units.Quantity`
+            The energy at which the statistical error is smallest.
+        """
+        dnde, dnde_error = self.evaluate_error(energy)
```

Comment:
why not the min of `dnde_error` directly ?

Reply:
This will not work, taking for example the `PowerLawSpectralModel`, the argmin  of dnde_error, would simply be the last energy bin in the plot (as we have a curve which is constantly decreasing.)

Reply:
I meant that the function  can be replaced by 

```
dnde, dnde_error = self.evaluate_error(energy)
energy[np.argmin(dnde_error/dnde)]
```
This will automatically include the condition `y_lo > 0`

Reply:
Apologies, I see what you mean. 

Good pick up, happy to implement it in this way! Thanks @AtreyeeS